src/planner_test.cc

```diff
@@ -170,6 +173,40 @@ class PlannerHarness {
   Planner *planner_;
 };
 
+template <typename T>
+static inline T fabs_accel(const T v0, const T v1, const int s) {
+  const T accel = (v1 * v1 - v0 * v0) / (2 * s);
+  return fabs(accel);
+}
+
+// Verify that on average, the profile satisfies
+// the configured constraints both for speed and acceleration.
+static void VerifySegmentConstraints(const MachineControlConfig &config,
+                                     const LinearSegmentSteps &segment,
+                                     const float epsilon = 1e-3) {
+  // Real world coordinates
+  const float axes[3] = {segment.steps[AXIS_X] / config.steps_per_mm[AXIS_X],
+                         segment.steps[AXIS_Y] / config.steps_per_mm[AXIS_Y],
+                         segment.steps[AXIS_Z] / config.steps_per_mm[AXIS_Z]};
+
+  const float hypotenuse =
+    sqrtf(axes[0] * axes[0] + axes[1] * axes[1] + axes[2] * axes[2]);
+
+  for (int i = 0; i <= AXIS_Z; ++i) {
+    const GCodeParserAxis axis = (GCodeParserAxis)i;
+    if (segment.steps[axis]) {
+      const float accel =
+        fabs_accel(segment.v0, segment.v1, segment.steps[axis]) * axes[i] /
+        (config.steps_per_mm[axis] * hypotenuse);
+      const float max_accel = config.acceleration[axis];
+      EXPECT_TRUE(fabs(accel - max_accel) < epsilon)
```

Comment:
there is also an `EXPECT_NEAR()` macro in google test.

Reply:
(solved using EXPECT_NEAR_REL)